### ISSUE TITLE
Transport nupkgs are symbol packages, which sometimes fail to be auto-converted to lib packages

### ISSUE THREAD

**Issue by dagood** [dotnet/coreclr#13798]
This is affecting Core-Setup official builds, which rely on the transport packages from CoreCLR.

CoreCLR pushes the transport packages to MyGet. Because the packages have pdbs and a src/ directory, they are detected as symbol packages. MyGet has a way to automatically convert symbol packages to library packages so they show up in the feed and are downloadable, but it doesn't always work. This causes affected packages not to show up in the gallery or feed, only available for download via direct URL.

Example: CoreCLR master `transport.runtime.win-arm64.Microsoft.NETCore.Runtime.CoreCLR` `2.1.0-preview2-25705-02`. It's [listed in dotnet/versions](https://github.com/dotnet/versions/blob/17a60c232e63766fff29109aabed9bddba46f19f/build-info/dotnet/coreclr/master/Last_Build_Packages.txt#L91) and available for [direct download as a symbol package](https://dotnet.myget.org/F/dotnet-core/symbols/transport.runtime.win-arm64.Microsoft.NETCore.Runtime.CoreCLR/2.1.0-preview2-25705-02), but doesn't show up in [the gallery page](https://dotnet.myget.org/feed/dotnet-core/package/nuget/transport.runtime.win-arm64.Microsoft.NETCore.Runtime.CoreCLR).

The solution is to change from `src/` to `sources/`, like CoreFX, to avoid detection as a symbol package and stop relying on MyGet's automatic conversion. It looks like CoreCLR only has `src/_._`--if so, removing the dir altogether should work too.

It looks like BuildTools is currently responsible for `src/_._`: [Packaging.targets](https://github.com/dotnet/buildtools/blob/6a8a2bc7a1be14b019fdbd96d6ce5e6a06d3867d/src/Microsoft.DotNet.Build.Tasks.Packaging/src/PackageFiles/Packaging.targets#L493-L498). I can't quite tell if that's somehow overridable from CoreCLR--I'm not familiar with how the transport packages are created.

**Comment by dagood** [comment]
@wtgodbe Do you think this is something you can fix?

/cc @chcosta

**Comment by wtgodbe** [comment]
@dagood do you mean removing the `src` directory from the transport package?

**Comment by dagood** [comment]
Yeah, I suggested calling it `sources` instead, but since there's nothing inside, just removing `src` works. I'm thinking it might be tricky because the logic for a lot of this stuff is in BuildTools.

**Comment by RussKeldorph** [comment]
@weshaggard @chcosta Can you comment on the right way to fix this--assuming it's still a problem?

**Comment by weshaggard** [comment]
Renaming or removing the src folder would seem to be the right way to fix this. We definitely still see this issue get hit from time to time so I think it is worth fixing these transport packages to make them not look like symbol packages.

**Comment by RussKeldorph** [comment]
@weshaggard Does that imply a fix to buildtools?  Can we rename `src` to `sources` [here](https://github.com/dotnet/buildtools/blob/2bd3c49e2ace5f8d9130ff07d4c7cefd033f7e64/src/Microsoft.DotNet.Build.Tasks.Packaging/src/PackageFiles/Packaging.targets#L497) or do we need to make it configurable so CoreCLR can override it?

**Comment by RussKeldorph** [comment]
@weshaggard Oops I misread: should we add an option to have CoreCLR opt out of this buildtools behavior or can we remove it altogether?

**Comment by weshaggard** [comment]
@dagood would be in a better place to answer that question, but if we may need to provide an opt-out in buildtools if one doesn't already exist.

**Comment by dagood** [comment]
I think BuildTools needs new functionality. Maybe some logic starting from this part of the code to exclude anything in `src/` (or put it in `sources/`)?
https://github.com/dotnet/buildtools/blob/2bd3c49e2ace5f8d9130ff07d4c7cefd033f7e64/src/Microsoft.DotNet.Build.Tasks.Packaging/src/NuGetPack.cs#L171-L176

@chcosta might have some other ideas here.

**Comment by RussKeldorph** [comment]
@chcosta Do you have any suggestions?

**Comment by wtgodbe** [comment]
I think this one is less critical for ZBB, It hasn't been affecting builds. We can probably get away with pushing this to future, and maybe get it in for one of the later previews.
